### PR TITLE
Fileserver

### PR DESCRIPTION
**Pull request by nick-prat**
File server, used as {server}/file/{filename}. Limited to only being able to send files in or below it's directory

### CODE REVIEW

File: src/foam/nanos/http/FileServlet.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    String pathInfo = req.getPathInfo();
+    String filePath = pathInfo.substring(SERVLET_NAME.length() + 2);
+    String fileName = "";
+
+    String[] paths = filePath.split("/");
+    if(paths.length > 0) {
```
Comment (kgrgreer) [inline src/foam/nanos/http/FileServlet.java]: See FOAM style guideline. Formatting should be:
if ( paths.length > 0 ) {

File: src/foam/nanos/http/FileServlet.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    } catch (IOException ignore) {}
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    String pathInfo = req.getPathInfo();
+    String filePath = pathInfo.substring(SERVLET_NAME.length() + 2);
+    String fileName = "";
+
+    String[] paths = filePath.split("/");
```
Comment (kgrgreer) [inline src/foam/nanos/http/FileServlet.java:58]: If I insert "../" into the path, does it let me read from a parent/ancestor directory?

File: src/foam/nanos/http/FileServlet.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class FileServlet
+    extends HttpServlet
+{
+
+  public static final String SERVLET_NAME = "file";
```
Comment (kgrgreer) [inline src/foam/nanos/http/FileServlet.java]: s/file/static/

File: src/foam/nanos/http/FileServlet.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    try {
+      File srcFile = new File(filePath);
+
+      String path = srcFile.getAbsolutePath();
+      String cwd = System.getProperty("user.dir");
+
+      if ( srcFile.isFile() && srcFile.canRead() && cwd.equals(path.substring(0, cwd.length())) ) {
+        resp.setContentType("application/octet-stream");
```
Comment (kgrgreer) [inline src/foam/nanos/http/FileServlet.java]: I think the content type should be set based on the extension of the file requested. Add a map to constants: which includes the types we're likely to use (ie. .js, .gif, .xvg, .png, .csv, .txt, .html, etc.)

See: http://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types

File: src/foam/nanos/http/FileServlet.java (hunk 2 above)
Comment (nick-prat) [reply src/foam/nanos/http/FileServlet.java:58]: It seems like `..` messes up the `URI.getPath` call in `HttpHandler` and it doesn't even reach the `FileServlet`

File: src/foam/nanos/http/FileServlet.java (hunk 3 above)
Comment (nick-prat) [reply src/foam/nanos/http/FileServlet.java]: Servlet's can't contain a / in their name in the way it's setup currently

Comment (kgrgreer) [reply src/foam/nanos/http/FileServlet.java]: s/file/static/ means: replace "file" with "static"

It's a short-hand programmer notation.

File: src/foam/nanos/http/FileServlet.java (hunk 2 above)
Comment (kgrgreer) [reply src/foam/nanos/http/FileServlet.java:58]: Even if the ../ is after the servlet name?  So /file/../foo.txt

**Pull request merged**